app/src/main/java/com/lightricks/feedexercise/data/FeedRepository.kt

```diff
@@ -1,9 +1,43 @@
 package com.lightricks.feedexercise.data
 
+import com.lightricks.feedexercise.domain.FeedError
+import com.lightricks.feedexercise.domain.FeedSuccess
+import com.lightricks.feedexercise.util.result.Result
+import kotlinx.coroutines.async
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+
 /**
  * This is our data layer abstraction. Users of this class don't need to know
  * where the data actually comes from (network, database or somewhere else).
  */
-class FeedRepository {
-    //todo: implement
+interface FeedRepository {
+    suspend fun getFeed(): Flow<Result<FeedError, FeedSuccess>>
+}
+
+internal class FeedRepositoryImpl(
+    private val localFeedDataSource: FeedDataSource,
+    private val localFeedDataStore: FeedDataStore,
+    private val remoteFeedDataSource: FeedDataSource
```

Comment:
IMO having the remote and local source implementing the same interface make thing hard to understand which of the two is the source of truth.

Reply:
About the local feed store/source, since at the moment it is only used by this class as read and write then I would pass the data source as is with the two functionalities without abstraction.

Reply:
As we discussed - I will use one implementor for store and source interfaces